php-checks/src/main/java/org/sonar/php/checks/utils/CheckUtils.java

```diff
@@ -175,5 +190,72 @@ private static void appendChild(StringBuilder sb, @Nullable SyntaxToken prevToke
   private static boolean isSpaceRequired(SyntaxToken prevToken, SyntaxToken token) {
     return (token.line() > prevToken.line()) || (prevToken.column() + prevToken.text().length() < token.column());
   }
+  
+  public static List<Kind> CONTAIN_STATEMENTS = ImmutableList.of(
+    Kind.SCRIPT,
+    Kind.BLOCK,
+    Kind.CASE_CLAUSE,
+    Kind.DEFAULT_CLAUSE,
+    Kind.DECLARE_STATEMENT,
+    Kind.IF_STATEMENT,
+    Kind.ALTRENATIVE_IF_STATEMENT,
+    Kind.ELSE_CLAUSE,
+    Kind.ALTERNATIVE_ELSE_CLAUSE,
+    Kind.ELSEIF_CLAUSE,
+    Kind.ALTERNATIVE_ELSEIF_CLAUSE,
+    Kind.FOREACH_STATEMENT,
+    Kind.ALTERNATIVE_FOREACH_STATEMENT,
+    Kind.FOR_STATEMENT,
+    Kind.ALTERNATIVE_FOR_STATEMENT,
+    Kind.NAMESPACE_STATEMENT,
+    Kind.WHILE_STATEMENT);
+
+  public static List<StatementTree> getStatements(Tree tree) {
```

Comment:
@pynicolas Indeed... but it's not so easy. What is list is empty? Null or instance of `StatementsTree` with empty list? 
Null will complicate work (as usual, you should check for null every time) and empty list will mean empty Tree which has no first token or line. So our `PHPTree#getLine()` might fail.